### PR TITLE
use newly added ODBC SQLPrimary/ForeignKeys API

### PR DESCRIPTION
**Pull request by overcoil**
internal ref: PECO-275

This change requires an ODBC driver with IC/PK/FK support: 
-Windows version 2.6.29+ (or 2.6.27 preview)
-MacOS version 2.6.29+

### CODE REVIEW

**Review by yunbodeng-db — APPROVED** [review]
lgtm to me but please allow others to take a look too.
there seems to be no easy way to switch off such as a ui configuration?

**Comment by moderakh** [comment]
> lgtm to me but please allow others to take a look too. there seems to be no easy way to switch off such as a ui configuration?

@yunbodeng-db @overcoil why do we need to disable the feature? is there a known issue?

**Comment by overcoil** [comment]
@moderakh @yunbodeng-db There is no *tidy* way to disable this feature in the field. If we are paranoid about this, I can put together a copy of the prior connector (which suppresses SQLPrimary/ForeignKeys) along with a writeup for: 1) disabling the (by-then packed in) Tableau connector; 2) installing/activating the prior connector.

**Comment by yunbodeng-db** [comment]
> > lgtm to me but please allow others to take a look too. there seems to be no easy way to switch off such as a ui configuration?
> 
> @yunbodeng-db @overcoil why do we need to disable the feature?

It's opposite. We are re-enabling them.

**Comment by yunbodeng-db** [comment]
> @moderakh @yunbodeng-db There is no tidy way to disable this feature in the field. If we are paranoid about this, I can put together a copy of the prior connector (which suppresses SQLPrimary/ForeignKeys) along with a writeup for: 1) disabling the (by-then packed in) Tableau connector; 2) installing/activating the prior connector.

It'd help. Do you know if it's easy to do?

**Comment by yunbodeng-db** [comment]
Now it occurs to me that ODBC is not bundled with Tableau release. Should the flags change be version sensitive? Is there a way to check it will only be changed if the ODBC driver's version is above such and such?

**Review by moderakh — APPROVED** [review]
LGTM, just two questions/comments bellow:

Tableau and ODBC gets installed/upgraded independently.

Tableau has auto upgrade feature which can bring in this change. However ODBC driver does not get upgraded automatically. We should add to our docs that odbc driver has to be upgraded to benefit from this feature.

Also we should validate that existing scenarios (except informational constraints) when using newer connector (with this feature enabled) + older odbc driver still work without any issue.

**Review by bogdanghit — APPROVED** [review]
LGTM.

Have we tested backwards compatibility with old drivers?

File: tableau-databricks/manifest.xml [hunk 1]
```diff
@@ -56,8 +56,8 @@ limitations under the License.
         <customization name='CAP_ODBC_FETCH_CONTINUE_ON_ERROR' value='no' />
         <customization name='CAP_ODBC_FORCE_SINGLE_ROW_BINDING' value='no' />
         <customization name='CAP_ODBC_METADATA_STRING_LENGTH_UNKNOWN' value='no' />
-        <customization name='CAP_ODBC_METADATA_SUPPRESS_SQLFOREIGNKEYS_API' value='yes' />
-        <customization name='CAP_ODBC_METADATA_SUPPRESS_SQLPRIMARYKEYS_API' value='yes' />
+        <customization name='CAP_ODBC_METADATA_SUPPRESS_SQLFOREIGNKEYS_API' value='no' />
+        <customization name='CAP_ODBC_METADATA_SUPPRESS_SQLPRIMARYKEYS_API' value='no' />
```
Comment (bogdanghit) [inline tableau-databricks/manifest.xml:60]: It may be good to document the sequence of metadata operations before and after the change.

**Pull request merged**